src/hooks/useHasChangedRoute.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react'
+import { useLocation } from 'react-router-dom'
+
+/**
+ * Custom hook that fires a callback when the route changes
+ * Normally, you can use `useEffect` on the pathname from useLocation, but this
+ * hook ignores the first call of useEffect
+ *
+ * @param {function} onRouteChange the callback fired when the route changes
+ */
+const useHasChangedRoute = (onRouteChange: () => void) => {
+  const { pathname } = useLocation()
+  const [currentRoute, setCurrentRoute] = useState(pathname)
```

Comment:
Wonder if this could be an instance var rather than state to prevent a re-render?

Reply:
Yeah, shouldn't make a difference - might be slightly better w/ instance var in that it would re-run unnecessarily. 
I initially wrote it w/ useRef - but changed it. Can use instanceVar.